client/connect-account-page/index.js

```diff
@@ -96,7 +100,50 @@ const ConnectPageOnboardingDisabled = () => (
 const ConnectPageOnboarding = () => {
 	const [ isSubmitted, setSubmitted ] = useState( false );
 
-	const handleSetup = () => {
+	const handleLocationCheck = ( availableCountries ) => {
+		// Reset the 'Set up' button state if merchant decided to stop
+		const whenDeclined = () => {
+			setSubmitted( false );
+		};
+		// Redirect the merchant if merchant decided to continue
+		const whenConfirmed = () => {
+			window.location = wcpaySettings.connect.url;
+		};
+
+		// Populate translated list of supported countries we want to render in the modal window.
+		const countries = Object.values( availableCountries )
+			.map( ( country ) => {
+				// eslint-disable-next-line @wordpress/i18n-no-variables
+				return __( country, 'woocommerce-payments' );
+			} )
+			.sort()
+			.map( ( country ) => {
+				return { title: country };
+			} );
+
+		const container = document.createElement( 'div' );
+		container.id = 'wcpay-onboarding-location-check-container';
+		document.body.appendChild( container );
+		ReactDOM.render(
+			<OnboardingLocationCheckModal
+				countries={ countries }
+				whenDeclined={ whenDeclined }
+				whenConfirmed={ whenConfirmed }
```

Comment:
**Nitpick:** I'd probably use the following naming convention for these callback props/functions:

```
onDeclined={ handleDeclined }
onConfirmed={ handleConfirmed }
```

Please ignore me if the current implementation is a standard a8c/Woo naming convention 😄

Reply:
Good catch, the common naming for callcbacks indeed goes the suggested one. Updated in beb61e1b5b65dfacf422e59fad197ce8880973cb